src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= objects.length) {
+            this.grow();
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size >= objects.length) {
+            this.grow();
+        }
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index: " + index + "size: " + size);
+        }
+        if (index <= size) {
+            Object[] newObjectsArray = new Object[objects.length + 1];
+            System.arraycopy(objects, 0, newObjectsArray, 0, index);
+            newObjectsArray[index] = value;
+            System.arraycopy(objects, index, newObjectsArray, index + 1, objects.length - index);
+            objects = newObjectsArray;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] newObjectsArray;
+        if (list != null) {
+            int additionalSize = list.size();
+            if (additionalSize > 0) {
+                newObjectsArray = new Object[size + additionalSize];
+                System.arraycopy(objects, 0, newObjectsArray, 0, size);
+                for (int i = 0; i < additionalSize; i++) {
+                    newObjectsArray[size + i] = list.get(i);
+                }
+                objects = newObjectsArray;
+            }
+            size += additionalSize;
+        }
```

Comment:
could we use our method `add(T value)` for every element in the list?

Reply:
Thank you) 
it's really much more simply